### PR TITLE
[libyaml] Fix linkage in non-Windows systems

### PR DESCRIPTION
**Pull request by AlexAltea**
Fixes issue #10789.

### CODE REVIEW

File: ports/libyaml/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,16 @@
 include(vcpkg_common_functions)
```
Comment (Cheney-W) [inline ports/libyaml/portfile.cmake:1]: `include(vcpkg_common_functions)` is deprecated. Please remove it and the blank line after it.

**Review by Cheney-W — CHANGES_REQUESTED** [review]
Thanks for your PR! 
1. Could you please modify the Version: 0.2.2-2 in `vcpkg/ports/libyaml/CONTROL` to Version: 0.2.2-3? 
2. Could you please modify `configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/libyaml/copyright COPYONLY)` in `vcpkg/ports/libyaml/portfile.cmake` to `configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)`?

**Comment by AlexAltea** [comment]
@Cheney-W Fixed!

**Review by Cheney-W — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**